liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateViewGenerator.java

```diff
@@ -94,7 +94,9 @@ public Sql[] generateSql(CreateViewStatement statement, Database database, SqlGe
                 CatalogAndSchema schema = new CatalogAndSchema(statement.getCatalogName(), statement.getSchemaName()).customize(database);
                 sql.add(new UnparsedSql("IF NOT EXISTS (SELECT * FROM sys.views WHERE object_id = OBJECT_ID(N'[" + schema.getSchemaName() + "].[" + statement.getViewName() + "]'))\n" +
                         "    EXEC sp_executesql N'CREATE VIEW [" + schema.getSchemaName() + "].[" + statement.getViewName() + "] AS SELECT ''This is a code stub which will be replaced by an Alter Statement'' as [code_stub]'"));
-                viewDefinition.replaceIfExists("CREATE", "ALTER");
+                if (viewDefinition.contains("CREATE") && (!viewDefinition.contains("OR") || !viewDefinition.contains("ALTER"))) {
+                    viewDefinition.replace("CREATE", "ALTER");
```

Comment:
Same here

Reply:
I added unit tests and found one possible problem which could happen and fixed it